File: src/main/java/no/nav/pensjon/innsyn/server/servlet/ProbeServlet.java
package no.nav.pensjon.innsyn.server.servlet;

import javax.servlet.ServletOutputStream;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import java.io.IOException;

@WebServlet(name = "ProbeServlet", urlPatterns = {"/isReady", "/isAlive"}, loadOnStartup = 1)
public class ProbeServlet extends HttpServlet {

    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
        response.setStatus(200);
        response.setContentType("application/json");
        try (ServletOutputStream out = response.getOutputStream()) {
            out.write("{\"status\":\"UP\"}".getBytes());
            out.flush();
        }
    }
}

File: src/main/kotlin/no/nav/pensjon/innsyn/controller/ProbeServlet.kt
package no.nav.pensjon.innsyn.controller

import no.nav.security.token.support.core.api.Unprotected
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController
import javax.servlet.http.HttpServlet

@Unprotected
@RestController
class ProbeServlet : HttpServlet() {
    @get:GetMapping("/isAlive")
    val isAlive: Nothing? = null

    @get:GetMapping("/isReady")
    val isReady: Nothing? = null
}